tools/congestion-model/src/evaluation/user_experience.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{Model, Round, Transaction, TransactionStatus};
+
+#[derive(Debug, Clone)]
+pub struct UserExperience {
+    pub successful_tx_delay_avg: Round,
+    pub successful_tx_delay_mean: Round,
+    pub successful_tx_delay_90th_percentile: Round,
+    pub rejected_tx_delay_avg: Round,
+    pub rejected_tx_delay_mean: Round,
+    pub rejected_tx_delay_90th_percentile: Round,
+    pub unresolved_transactions: u64,
+}
+
+impl Model {
+    pub fn user_experience(&self) -> UserExperience {
+        let mut successful_delays = vec![];
+        let mut rejected_delays = vec![];
+        let mut unresolved_transactions = 0;
+        for tx in self.transactions.all_transactions() {
+            match tx.status() {
+                TransactionStatus::Init | TransactionStatus::Pending => {
+                    unresolved_transactions += 1
+                }
+                TransactionStatus::Failed => rejected_delays.push(tx.delay()),
+                TransactionStatus::FinishedSuccess => successful_delays.push(tx.delay()),
+            }
+        }
+
+        successful_delays.sort();
+        rejected_delays.sort();
+
+        UserExperience {
+            successful_tx_delay_avg: avg(&successful_delays),
+            successful_tx_delay_mean: percentile(&successful_delays, 50),
```

Comment:
median is the 50th percentile, mean is typically the same as average

Reply:
yeah, brainfart :upside_down_face: 
thanks!